### PR TITLE
testing/py-unittest2 + 3 x depends

### PR DESCRIPTION
**Pull request by itoffshore**
* py-contextlib2
* py-traceback2
* py-linecache2
* py-unittest2

### CODE REVIEW

**Comment by itoffshore** [comment]
rebased with:

```
build()
...
check()
...
package()
```

File: testing/py-unittest2/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Contributor: Stuart Cardall <developer@it-offshore.co.uk>
+# Maintainer: Stuart Cardall <developer@it-offshore.co.uk>
+pkgname=py-unittest2
```
Comment (jirutka) [inline testing/py-unittest2/APKBUILD]: This is Python2-only, so it should be named `py2-unittest2`.

File: testing/py-unittest2/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+_pkgname=${pkgname#py-*}
+pkgver=1.1.0
+pkgrel=0
+pkgdesc="Backport of the Python3 enhanced unittest testing framework"
+url="https://pypi.python.org/pypi/unittest2"
+arch="noarch"
+license="PSF"
+depends="python py-six py-argparse py-traceback2 py-linecache2"
```
Comment (jirutka) [inline testing/py-unittest2/APKBUILD]: Please don’t declare versionless python packages as dependencies, it should be python2, py2-six, py2-argparse, …

File: testing/py-unittest2/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+makedepends="python2-dev py-setuptools"
+source="$pkgname-$pkgver.tar.gz::http://http.debian.net/debian/pool/main/u/$_pkgname/${_pkgname}_${pkgver}.orig.tar.gz"
+builddir="$srcdir/$_pkgname-$pkgver"
+
+build() {
+        cd "$builddir"
+        python setup.py build
```
Comment (jirutka) [inline testing/py-unittest2/APKBUILD]: Please don’t rely on `python` symlink, use explicitly `python2`, or `python3`.

File: testing/py-traceback2/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Contributor: Stuart Cardall <developer@it-offshore.co.uk>
+# Maintainer: Stuart Cardall <developer@it-offshore.co.uk>
+pkgname=py-traceback2
```
Comment (jirutka) [inline testing/py-traceback2/APKBUILD]: This module supports both Python 2 (legacy) and 3, so it must be packaged for Python 3 only (py3-traceback2), or for both Python 2 and 3 (py-traceback2 with subpackages py2-traceback2 and py3-traceback2). If there’s no specific reason for supporting Legacy Python, I’d prefer py3 only package (it’s simpler).

Comment (itoffshore) [inline testing/py-traceback2/APKBUILD]: `python2` versions all of these 4 depends are needed so `py-bandit` can be added with a `check()` 

(which will finally allow me to fix another `python` package with a `check()` using `py-bandit`)

**Comment by itoffshore** [comment]
all the names / depends explicitly set to `py2` for:
* py2-contextlib2
* py2-linecache2
* py2-unittest2

py-traceback2 now with `python2` & `python3` versions

File: testing/py-traceback2/APKBUILD (hunk 4 above)
Comment (itoffshore) [reply testing/py-traceback2/APKBUILD]: any reason why this is still not pulled ???

File: testing/py-traceback2/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        python2 -m unittest2 discover
+        python3 -m unittest2 discover
+}
+
+package() {
+	mkdir -p "$pkgdir"
+}
+
+_py2() {
+	replaces="$pkgname"
```
Comment (jirutka) [inline testing/py-traceback2/APKBUILD]: It’s a new aport, so `replaces` is not needed.

**Pull request closed without merging**